Fails with

```
main.cpp:3:43: fatal error: asemantools/asemanapplication.h: No such file or directory
 #include "asemantools/asemanapplication.h"
```

--- a/pkgs/applications/networking/instant-messengers/telegram/cutegram/default.nix
+++ b/pkgs/applications/networking/instant-messengers/telegram/cutegram/default.nix
@@ -1,31 +1,37 @@
-{ stdenv, fetchFromGitHub
-, qtbase, qtquick1, qtmultimedia, qtquickcontrols, qtgraphicaleffects, makeQtWrapper
-, telegram-qml, libqtelegram-aseman-edition }:
+{ stdenv, fetchgit
+, qtbase, qtmultimedia, qtquick1, qtquickcontrols, qtgraphicaleffects
+, telegram-qml, libqtelegram-aseman-edition
+, gst_plugins_base, gst_plugins_good, gst_plugins_bad, gst_plugins_ugly
+, makeQtWrapper }:
 
 stdenv.mkDerivation rec {
-  name = "cutegram-${version}";
-  version = "2.7.0-stable";
+  name = "cutegram-${meta.version}";
 
-  src = fetchFromGitHub {
-    owner = "Aseman-Land";
-    repo = "Cutegram";
-    rev = "v${version}";
-    sha256 = "0qhy30gb8zdrphz1b7zcnv8hmm5fd5qwlvrg7wpsh3hk5niz3zxk";
+  src = fetchgit {
+    url = "https://github.com/Aseman-Land/Cutegram.git";
+    rev = "1dbe2792fb5a1760339379907f906e236c09db84";
+    sha256 = "080153bpa92jpi0zdrfajrn0yqy3jp8m4704sirbz46dv7471rzl";
   };
-  # TODO appindicator, for system tray plugin
-  buildInputs = [ qtbase qtquick1 qtmultimedia qtquickcontrols qtgraphicaleffects telegram-qml libqtelegram-aseman-edition ];
+
+  buildInputs =
+  [ qtbase qtmultimedia qtquick1 qtquickcontrols qtgraphicaleffects
+    telegram-qml libqtelegram-aseman-edition
+    gst_plugins_base gst_plugins_good gst_plugins_bad gst_plugins_ugly ];
   nativeBuildInputs = [ makeQtWrapper ];
   enableParallelBuild = true;
 
-  fixupPhase = "wrapQtProgram $out/bin/cutegram";
-
   configurePhase = "qmake -r PREFIX=$out";
 
+  fixupPhase = "wrapQtProgram $out/bin/cutegram";
+
   meta = with stdenv.lib; {
+    version = "2.7.1";
     description = "Telegram client forked from sigram";
     homepage = "http://aseman.co/en/products/cutegram/";
     license = licenses.gpl3;
-    maintainers = [ maintainers.profpatsch ];
+    maintainers = with maintainers; [ profpatsch AndersonTorres ];
   };
-
 }
+#TODO: appindicator, for system tray plugin (by @profpatsch)
+
+

--- a/pkgs/applications/networking/instant-messengers/telegram/libqtelegram-aseman-edition/default.nix
+++ b/pkgs/applications/networking/instant-messengers/telegram/libqtelegram-aseman-edition/default.nix
@@ -2,14 +2,13 @@
 , qtbase, qtmultimedia, qtquick1 }:
 
 stdenv.mkDerivation rec {
-  name = "libqtelegram-aseman-edition-${version}";
-  version = "6.0";
+  name = "libqtelegram-aseman-edition-${meta.version}";
 
   src = fetchFromGitHub {
     owner = "Aseman-Land";
     repo = "libqtelegram-aseman-edition";
-    rev = "v${version}";
-    sha256 = "17hlxf43xwic8m06q3gwbxjpvz31ks6laffjw6ny98d45zfnfwra";
+    rev = "v${meta.version}-stable";
+    sha256 = "1pfd4pvh51639zk9shv1s4f6pf0ympnhar8a302vhrkga9i4cbx6";
   };
 
   buildInputs = [ qtbase qtmultimedia qtquick1 ];
@@ -25,6 +24,7 @@ stdenv.mkDerivation rec {
   '';
 
   meta = with stdenv.lib; {
+    version = "6.1";
     description = "A fork of libqtelegram by Aseman, using qmake";
     homepage = src.meta.homepage;
     license = stdenv.lib.licenses.gpl3;

--- a/pkgs/applications/networking/instant-messengers/telegram/telegram-qml/default.nix
+++ b/pkgs/applications/networking/instant-messengers/telegram/telegram-qml/default.nix
@@ -3,17 +3,16 @@
 , libqtelegram-aseman-edition }:
 
 stdenv.mkDerivation rec {
-  name = "telegram-qml-${version}";
-  version = "0.9.1-stable";
+  name = "telegram-qml-${meta.version}";
 
   src = fetchFromGitHub {
     owner = "Aseman-Land";
     repo = "TelegramQML";
-    rev = "v${version}";
-    sha256 = "077j06lfr6qccqv664hn0ln023xlh5cfm50kapjc2inapxj2yqmn";
+    rev = "v${meta.version}";
+    sha256 = "0j8vn845f2virvddk9yjbljy6vkr9ikyn6iy7hpj8nvr2xls3499";
   };
 
-  buildInputs = [ qtbase qtmultimedia qtquick1 libqtelegram-aseman-edition ];
+  propagatedBuildInputs = [ qtbase qtmultimedia qtquick1 libqtelegram-aseman-edition ];
   enableParallelBuild = true;
 
   patchPhase = ''
@@ -26,6 +25,7 @@ stdenv.mkDerivation rec {
   '';
 
   meta = with stdenv.lib; {
+    version = "0.9.2";
     description = "Telegram API tools for QtQml and Qml";
     homepage = src.meta.homepage;
     license = stdenv.lib.licenses.gpl3;

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11653,10 +11653,10 @@ let
   cutecom = callPackage ../tools/misc/cutecom { };
 
   cutegram =
-    let cp = qt5.callPackage;
-    in cp ../applications/networking/instant-messengers/telegram/cutegram rec {
-      libqtelegram-aseman-edition = cp ../applications/networking/instant-messengers/telegram/libqtelegram-aseman-edition { };
-      telegram-qml = cp ../applications/networking/instant-messengers/telegram/telegram-qml {
+    let callpkg = qt55.callPackage;
+    in callpkg ../applications/networking/instant-messengers/telegram/cutegram rec {
+      libqtelegram-aseman-edition = callpkg ../applications/networking/instant-messengers/telegram/libqtelegram-aseman-edition { };
+      telegram-qml = callpkg ../applications/networking/instant-messengers/telegram/telegram-qml {
         inherit libqtelegram-aseman-edition;
       };
     };

